crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -159,6 +196,71 @@ impl PendingTransactionsForAccount {
     fn highest_nonce(&self) -> Option<u32> {
         self.txs.last_key_value().map(|(nonce, _)| *nonce)
     }
+
+    /// Removes and returns transactions that exceed the balances in `available_balances`.
+    fn find_demotables(
+        &mut self,
+        mut available_balances: HashMap<IbcPrefixed, u128>,
+    ) -> Vec<TimemarkedTransaction> {
+        let mut split_at = 0;
+
+        'outer: for (nonce, tx) in &self.txs {
+            // ensure we have enough balance to cover inclusion
+            for (denom, cost) in tx.cost() {
+                if *cost == 0 {
+                    continue;
+                }
+                match available_balances.entry(*denom) {
+                    hash_map::Entry::Occupied(mut entry) => {
+                        // try to subtract cost, if not enough balance, do not include
+                        let current_balance = entry.get_mut();
+                        *current_balance = match current_balance.checked_sub(*cost) {
+                            None => break 'outer,
+                            Some(new_value) => new_value,
+                        };
+                    }
+                    hash_map::Entry::Vacant(_) => {
+                        // not enough balance, do not include
+                        break 'outer;
+                    }
+                }
+            }
+
+            split_at = nonce.saturating_add(1);
+        }
+
+        // return all keys higher than split target
+        self.txs.split_off(&split_at).into_values().collect()
+    }
+
+    /// Returns remaining balances after accounting for costs of contained transactions.
+    ///
+    /// Note: assumes that the balances in `current_account_balances` are large enough
+    /// to cover costs for contained transactions. Will log an error if this is not true
+    /// but will not fail.
+    fn get_remaining_balances(
+        &self,
+        mut remaining_account_balances: HashMap<IbcPrefixed, u128>,
```

Comment:
the fact that this takes an argument throws me off. Can there be a better name? Also, shouldn't this take `remaining_account_balances` as a `&mut`?

Reply:
renamed `subtract_contained_costs` and fixed the `&mut`!